forch/forchestrator.py

```diff
@@ -436,7 +440,15 @@ def _restore_states(self):
     def _restore_faucet_config(self, timestamp, config_hash):
         config_info, faucet_dps, _ = self._get_faucet_config()
         self._update_config_warning_varz()
-        assert config_hash == config_info['hashes'], 'config hash info does not match'
+
+        if config_hash == config_info['hashes']:
+            self._config_hash_retry = 0
+        else:
+            assert self._config_hash_retry < self._config_hash_max_retry, (
+                f'config hash info does not match after {self._config_hash_max_retry} retries')
+            self._config_hash_retry += 1
```

Comment:
how often does this happen?  What, in practical terms, does a value of "5" mean?  Is it periodic or 5 of some particular kind of event?

Reply:
I just ran Eng test again and analyzed the sequence of the events. I think this value is mostly about how offen there is an L2_EXPIRE event is received with a CONFIG_CHANGE event. And that is related to the number of devices and the timing of the expiry of the devices. For example, if every few seconds there is a device expires, it is more likely that a hash mismatch happens, because Faucet reloads the config every few seconds.

In Eng's test, I can see the highest retry is 3 with about 10 devices.

Reply:
So -- do you think a simple count of the number of hash failures is the right function?  Seems fairly fragile and dependent on the number of devices, etc... Seems like a better thing would be "X seconds after the last hash clash"?

Reply:
Changed to a time based...